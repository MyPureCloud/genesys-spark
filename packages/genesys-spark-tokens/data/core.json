{
    "spacing": {
        "4xs": {
            "value": "4px",
            "type": "spacing"
        },
        "3xs": {
            "value": "8px",
            "type": "spacing"
        },
        "2xs": {
            "value": "12px",
            "type": "spacing"
        },
        "xs": {
            "value": "16px",
            "type": "spacing"
        },
        "sm": {
            "value": "20px",
            "type": "spacing"
        },
        "md": {
            "value": "24px",
            "type": "spacing"
        },
        "lg": {
            "value": "32px",
            "type": "spacing"
        },
        "xl": {
            "value": "40px",
            "type": "spacing"
        },
        "2xl": {
            "value": "48px",
            "type": "spacing"
        }
    },
    "z_index": {
        "0": {
            "value": "0",
            "type": "zindex"
        },
        "1": {
            "value": "1",
            "type": "zindex"
        },
        "100": {
            "value": "100",
            "type": "zindex"
        },
        "200": {
            "value": "200",
            "type": "zindex"
        },
        "300": {
            "value": "300",
            "type": "zindex"
        },
        "400": {
            "value": "400",
            "type": "zindex"
        },
        "500": {
            "value": "500",
            "type": "zindex"
        },
        "600": {
            "value": "600",
            "type": "zindex"
        }
    },
    "border_radius": {
        "none": {
            "value": "0",
            "type": "borderRadius"
        },
        "sm": {
            "value": "4px",
            "type": "borderRadius"
        },
        "md": {
            "value": "8px",
            "type": "borderRadius"
        },
        "lg": {
            "value": "16px",
            "type": "borderRadius"
        },
        "xl": {
            "value": "20px",
            "type": "borderRadius"
        },
        "full": {
            "value": "100%",
            "type": "borderRadius"
        }
    },
    "color": {
        "white": {
            "value": "#ffffff",
            "type": "color"
        },
        "black": {
            "value": "#000000",
            "type": "color"
        },
        "transparent": {
            "value": "#00000000",
            "type": "color"
        },
        "cornflower": {
            "100": {
                "value": "#f2f2f3",
                "type": "color"
            },
            "200": {
                "value": "#e4e5e7",
                "type": "color"
            },
            "300": {
                "value": "#c6c8ce",
                "type": "color"
            },
            "400": {
                "value": "#a3a7b0",
                "type": "color"
            },
            "500": {
                "value": "#767c8d",
                "type": "color"
            },
            "600": {
                "value": "#626e84",
                "type": "color"
            },
            "700": {
                "value": "#3e5374",
                "type": "color"
            },
            "800": {
                "value": "#23395d",
                "type": "color"
            },
            "900": {
                "value": "#1b2c48",
                "type": "color"
            },
            "1000": {
                "value": "#0f1929",
                "type": "color"
            }
        },
        "islandaqua": {
            "100": {
                "value": "#EDF8F9",
                "type": "color"
            },
            "200": {
                "value": "#DAF0F4",
                "type": "color"
            },
            "300": {
                "value": "#B5E2E8",
                "type": "color"
            },
            "400": {
                "value": "#90D3DD",
                "type": "color"
            },
            "500": {
                "value": "#6BC5D1",
                "type": "color"
            },
            "600": {
                "value": "#46B6C6",
                "type": "color"
            },
            "700": {
                "value": "#2C808C",
                "type": "color"
            },
            "800": {
                "value": "#0C6D7E",
                "type": "color"
            },
            "900": {
                "value": "#115D71",
                "type": "color"
            },
            "1000": {
                "value": "#094052",
                "type": "color"
            }
        },
        "tomato": {
            "100": {
                "value": "#fef3f3",
                "type": "color"
            },
            "200": {
                "value": "#fee8e6",
                "type": "color"
            },
            "300": {
                "value": "#fecfcb",
                "type": "color"
            },
            "400": {
                "value": "#feb2ab",
                "type": "color"
            },
            "500": {
                "value": "#fe8f85",
                "type": "color"
            },
            "600": {
                "value": "#fe624d",
                "type": "color"
            },
            "700": {
                "value": "#e35744",
                "type": "color"
            },
            "800": {
                "value": "#c44b3b",
                "type": "color"
            },
            "900": {
                "value": "#a03d30",
                "type": "color"
            },
            "1000": {
                "value": "#712b22",
                "type": "color"
            }
        },
        "lavenderhaze": {
            "100": {
                "value": "#fbfcfe",
                "type": "color"
            },
            "200": {
                "value": "#f3f5fb",
                "type": "color"
            },
            "300": {
                "value": "#e8ebf7",
                "type": "color"
            },
            "400": {
                "value": "#d1d7ef",
                "type": "color"
            },
            "500": {
                "value": "#c6ceec",
                "type": "color"
            },
            "600": {
                "value": "#b5bfe6",
                "type": "color"
            },
            "700": {
                "value": "#a8b4e2",
                "type": "color"
            },
            "800": {
                "value": "#9ba9dd",
                "type": "color"
            },
            "900": {
                "value": "#7785b8",
                "type": "color"
            },
            "1000": {
                "value": "#626d97",
                "type": "color"
            }
        },
        "blueberry": {
            "50": {
                "value": "#EFF0FE",
                "type": "color"
            },
            "100": {
                "value": "#DEE1FD",
                "type": "color"
            },
            "200": {
                "value": "#CED2FC",
                "type": "color"
            },
            "300": {
                "value": "#9CA6F9",
                "type": "color"
            },
            "400": {
                "value": "#7B88F7",
                "type": "color"
            },
            "500": {
                "value": "#5A6AF5",
                "type": "color"
            },
            "600": {
                "value": "#4856CB",
                "type": "color"
            },
            "700": {
                "value": "#3742A2",
                "type": "color"
            },
            "800": {
                "value": "#252E78",
                "type": "color"
            },
            "900": {
                "value": "#1C2463",
                "type": "color"
            },
            "1000": {
                "value": "#141A4F",
                "type": "color"
            }
        },
        "mineral": {
            "100": {
                "value": "#DDEAF7",
                "type": "color"
            },
            "200": {
                "value": "#CDE0F4",
                "type": "color"
            },
            "300": {
                "value": "#BCD6F0",
                "type": "color"
            },
            "400": {
                "value": "#9AC1E8",
                "type": "color"
            },
            "500": {
                "value": "#79ADE1",
                "type": "color"
            },
            "600": {
                "value": "#5798D9",
                "type": "color"
            },
            "700": {
                "value": "#467AAE",
                "type": "color"
            },
            "800": {
                "value": "#345B82",
                "type": "color"
            },
            "900": {
                "value": "#233D57",
                "type": "color"
            },
            "1000": {
                "value": "#1A2E41",
                "type": "color"
            }
        },
        "plum": {
            "100": {
                "value": "#EEE3FF",
                "type": "color"
            },
            "200": {
                "value": "#E6D6FF",
                "type": "color"
            },
            "300": {
                "value": "#DEC8FF",
                "type": "color"
            },
            "400": {
                "value": "#CDACFF",
                "type": "color"
            },
            "500": {
                "value": "#BD91FF",
                "type": "color"
            },
            "600": {
                "value": "#AC75FF",
                "type": "color"
            },
            "700": {
                "value": "#8A5ECC",
                "type": "color"
            },
            "800": {
                "value": "#674699",
                "type": "color"
            },
            "900": {
                "value": "#452F66",
                "type": "color"
            },
            "1000": {
                "value": "#34234D",
                "type": "color"
            }
        },
        "raspberry": {
            "100": {
                "value": "#FADFF5",
                "type": "color"
            },
            "200": {
                "value": "#F7CFF0",
                "type": "color"
            },
            "300": {
                "value": "#F5BFEB",
                "type": "color"
            },
            "400": {
                "value": "#EF9EE1",
                "type": "color"
            },
            "500": {
                "value": "#EA7ED7",
                "type": "color"
            },
            "600": {
                "value": "#E55ECD",
                "type": "color"
            },
            "700": {
                "value": "#B74BA4",
                "type": "color"
            },
            "800": {
                "value": "#89387B",
                "type": "color"
            },
            "900": {
                "value": "#5C2652",
                "type": "color"
            },
            "1000": {
                "value": "#451C3E",
                "type": "color"
            }
        },
        "coral": {
            "100": {
                "value": "#FFDEE4",
                "type": "color"
            },
            "200": {
                "value": "#FFCED6",
                "type": "color"
            },
            "300": {
                "value": "#FFBEC9",
                "type": "color"
            },
            "400": {
                "value": "#FF9DAD",
                "type": "color"
            },
            "500": {
                "value": "#FF7D92",
                "type": "color"
            },
            "600": {
                "value": "#FF5C77",
                "type": "color"
            },
            "700": {
                "value": "#CC4A5F",
                "type": "color"
            },
            "800": {
                "value": "#993747",
                "type": "color"
            },
            "900": {
                "value": "#662530",
                "type": "color"
            },
            "1000": {
                "value": "#4D1C24",
                "type": "color"
            }
        },
        "mango": {
            "100": {
                "value": "#FFF4DC",
                "type": "color"
            },
            "200": {
                "value": "#FFEECA",
                "type": "color"
            },
            "300": {
                "value": "#FFE8B9",
                "type": "color"
            },
            "400": {
                "value": "#FFDD96",
                "type": "color"
            },
            "500": {
                "value": "#FFD173",
                "type": "color"
            },
            "600": {
                "value": "#FFC650",
                "type": "color"
            },
            "700": {
                "value": "#CC9E40",
                "type": "color"
            },
            "800": {
                "value": "#997730",
                "type": "color"
            },
            "900": {
                "value": "#664F20",
                "type": "color"
            },
            "1000": {
                "value": "#4D3B18",
                "type": "color"
            }
        },
        "pear": {
            "100": {
                "value": "#EDF5DF",
                "type": "color"
            },
            "200": {
                "value": "#E3EECC",
                "type": "color"
            },
            "300": {
                "value": "#D9E8BB",
                "type": "color"
            },
            "400": {
                "value": "#C6DD98",
                "type": "color"
            },
            "500": {
                "value": "#B3D176",
                "type": "color"
            },
            "600": {
                "value": "#A0C654",
                "type": "color"
            },
            "700": {
                "value": "#809E43",
                "type": "color"
            },
            "800": {
                "value": "#607732",
                "type": "color"
            },
            "900": {
                "value": "#404F22",
                "type": "color"
            },
            "1000": {
                "value": "#303B19",
                "type": "color"
            }
        },
        "jade": {
            "100": {
                "value": "#DFF5F0",
                "type": "color"
            },
            "200": {
                "value": "#CCEEE6",
                "type": "color"
            },
            "300": {
                "value": "#BBE8DE",
                "type": "color"
            },
            "400": {
                "value": "#98DDCD",
                "type": "color"
            },
            "500": {
                "value": "#76D1BC",
                "type": "color"
            },
            "600": {
                "value": "#54C6AB",
                "type": "color"
            },
            "700": {
                "value": "#439E89",
                "type": "color"
            },
            "800": {
                "value": "#327767",
                "type": "color"
            },
            "900": {
                "value": "#224F45",
                "type": "color"
            },
            "1000": {
                "value": "#193B33",
                "type": "color"
            }
        }
    },
    "font_family": {
        "body": {
            "value": "Noto Sans",
            "type": "fontFamilies"
        },
        "heading": {
            "value": "Urbanist",
            "type": "fontFamilies"
        }
    },
    "font_size": {
        "xs": {
            "value": "12px",
            "type": "fontSizes"
        },
        "sm": {
            "value": "14px",
            "type": "fontSizes"
        },
        "md": {
            "value": "16px",
            "type": "fontSizes"
        },
        "lg": {
            "value": "18px",
            "type": "fontSizes"
        },
        "xl": {
            "value": "24px",
            "type": "fontSizes"
        },
        "xxl": {
            "value": "36px",
            "type": "fontSizes"
        }
    },
    "font_weight": {
        "regular": {
            "value": "400",
            "type": "fontWeights"
        },
        "semi-bold": {
            "value": "600",
            "type": "fontWeights"
        },
        "bold": {
            "value": "700",
            "type": "fontWeights"
        }
    },
    "line_height": {
        "2xs": {
            "value": "16px",
            "type": "lineHeights"
        },
        "xs": {
            "value": "18px",
            "type": "lineHeights"
        },
        "sm": {
            "value": "20px",
            "type": "lineHeights"
        },
        "md": {
            "value": "24px",
            "type": "lineHeights"
        },
        "lg": {
            "value": "27px",
            "type": "lineHeights"
        },
        "xl": {
            "value": "32px",
            "type": "lineHeights"
        },
        "2xl": {
            "value": "44px",
            "type": "lineHeights"
        },
        "match_font_size": {
            "value": "1",
            "type": "lineHeights",
            "description": "This token is used for matching the line-height value to the font-size value on CSS; for example if you have an 8px font size property and you apply this line-height, the line-height will have the value of 8px."
        }
    },
    "typography": {
        "body": {
            "sm": {
                "value": "{typography.body.sm_regular}",
                "type": "typography"
            },
            "sm_bold": {
                "value": {
                    "fontFamily": "{font_family.body}",
                    "lineHeight": "{line_height.xs}",
                    "fontSize": "{font_size.xs}",
                    "fontWeight": "{font_weight.bold}"
                },
                "type": "typography"
            },
            "sm_semi_bold": {
                "value": {
                    "fontFamily": "{font_family.body}",
                    "lineHeight": "{line_height.xs}",
                    "fontSize": "{font_size.xs}",
                    "fontWeight": "{font_weight.semi-bold}"
                },
                "type": "typography"
            },
            "sm_regular": {
                "value": {
                    "fontFamily": "{font_family.body}",
                    "lineHeight": "{line_height.xs}",
                    "fontSize": "{font_size.xs}",
                    "fontWeight": "{font_weight.regular}"
                },
                "type": "typography"
            },
            "md": {
                "value": "{typography.body.md_regular}",
                "type": "typography"
            },
            "md_regular": {
                "value": {
                    "fontFamily": "{font_family.body}",
                    "lineHeight": "{line_height.sm}",
                    "fontSize": "{font_size.sm}",
                    "fontWeight": "{font_weight.regular}"
                },
                "type": "typography"
            },
            "md_semi_bold": {
                "value": {
                    "fontFamily": "{font_family.body}",
                    "lineHeight": "{line_height.sm}",
                    "fontSize": "{font_size.sm}",
                    "fontWeight": "{font_weight.semi-bold}"
                },
                "type": "typography"
            },
            "md_bold": {
                "value": {
                    "fontFamily": "{font_family.body}",
                    "lineHeight": "{line_height.sm}",
                    "fontSize": "{font_size.sm}",
                    "fontWeight": "{font_weight.bold}"
                },
                "type": "typography"
            },
            "lg": {
                "value": "{typography.body.lg_regular}",
                "type": "typography"
            },
            "lg_regular": {
                "value": {
                    "fontFamily": "{font_family.body}",
                    "lineHeight": "{line_height.md}",
                    "fontSize": "{font_size.md}",
                    "fontWeight": "{font_weight.regular}"
                },
                "type": "typography"
            },
            "lg_semi_bold": {
                "value": {
                    "fontFamily": "{font_family.body}",
                    "lineHeight": "{line_height.md}",
                    "fontSize": "{font_size.md}",
                    "fontWeight": "{font_weight.semi-bold}"
                },
                "type": "typography"
            },
            "lg_bold": {
                "value": {
                    "fontFamily": "{font_family.body}",
                    "lineHeight": "{line_height.md}",
                    "fontSize": "{font_size.md}",
                    "fontWeight": "{font_weight.bold}"
                },
                "type": "typography"
            }
        },
        "heading": {
            "xs": {
                "value": "{typography.heading.xs_bold}",
                "type": "typography"
            },
            "xs_bold": {
                "value": {
                    "fontFamily": "{font_family.heading}",
                    "lineHeight": "{line_height.md}",
                    "fontSize": "{font_size.sm}",
                    "fontWeight": "{font_weight.bold}"
                },
                "type": "typography"
            },
            "xs_semi_bold": {
                "value": {
                    "fontFamily": "{font_family.heading}",
                    "lineHeight": "{line_height.md}",
                    "fontSize": "{font_size.sm}",
                    "fontWeight": "{font_weight.semi-bold}"
                },
                "type": "typography"
            },
            "sm": {
                "value": "{typography.heading.sm_bold}",
                "type": "typography"
            },
            "sm_bold": {
                "value": {
                    "fontFamily": "{font_family.heading}",
                    "lineHeight": "{line_height.md}",
                    "fontSize": "{font_size.md}",
                    "fontWeight": "{font_weight.bold}"
                },
                "type": "typography"
            },
            "sm_semi_bold": {
                "value": {
                    "fontFamily": "{font_family.heading}",
                    "lineHeight": "{line_height.md}",
                    "fontSize": "{font_size.md}",
                    "fontWeight": "{font_weight.semi-bold}"
                },
                "type": "typography"
            },
            "md": {
                "value": "{typography.heading.md_bold}",
                "type": "typography"
            },
            "md_bold": {
                "value": {
                    "fontFamily": "{font_family.heading}",
                    "lineHeight": "{line_height.lg}",
                    "fontSize": "{font_size.lg}",
                    "fontWeight": "{font_weight.bold}"
                },
                "type": "typography"
            },
            "md_semi_bold": {
                "value": {
                    "fontFamily": "{font_family.heading}",
                    "lineHeight": "{line_height.lg}",
                    "fontSize": "{font_size.lg}",
                    "fontWeight": "{font_weight.semi-bold}"
                },
                "type": "typography"
            },
            "lg": {
                "value": "{typography.heading.lg_bold}",
                "type": "typography"
            },
            "lg_bold": {
                "value": {
                    "fontFamily": "{font_family.heading}",
                    "lineHeight": "{line_height.xl}",
                    "fontSize": "{font_size.xl}",
                    "fontWeight": "{font_weight.bold}"
                },
                "type": "typography"
            },
            "lg_semi_bold": {
                "value": {
                    "fontFamily": "{font_family.heading}",
                    "lineHeight": "{line_height.xl}",
                    "fontSize": "{font_size.xl}",
                    "fontWeight": "{font_weight.semi-bold}"
                },
                "type": "typography"
            },
            "xl": {
                "value": "{typography.heading.xl_bold}",
                "type": "typography"
            },
            "xl_bold": {
                "value": {
                    "fontFamily": "{font_family.heading}",
                    "lineHeight": "{line_height.2xl}",
                    "fontSize": "{font_size.xxl}",
                    "fontWeight": "{font_weight.bold}"
                },
                "type": "typography"
            },
            "xl_semi_bold": {
                "value": {
                    "fontFamily": "{font_family.heading}",
                    "lineHeight": "{line_height.2xl}",
                    "fontSize": "{font_size.xxl}",
                    "fontWeight": "{font_weight.semi-bold}"
                },
                "type": "typography"
            },
            "overline": {
                "value": {
                    "fontFamily": "{font_family.heading}",
                    "fontWeight": "{font_weight.semi-bold}",
                    "lineHeight": "{line_height.2xs}",
                    "fontSize": "{font_size.xs}",
                    "letterSpacing": "1.5px",
                    "textCase": "uppercase"
                },
                "type": "typography"
            }
        },
        "subheading": {
            "value": "{typography.subheading_bold}",
            "type": "typography"
        },
        "subheading_bold": {
            "value": {
                "fontFamily": "{font_family.heading}",
                "fontWeight": "{font_weight.bold}",
                "lineHeight": "{line_height.sm}",
                "fontSize": "{font_size.sm}"
            },
            "type": "typography"
        },
        "subheading_semi_bold": {
            "value": {
                "fontFamily": "{font_family.heading}",
                "fontWeight": "{font_weight.semi-bold}",
                "lineHeight": "{line_height.sm}",
                "fontSize": "{font_size.sm}"
            },
            "type": "typography"
        },
        "subheading_regular": {
            "value": {
                "fontFamily": "{font_family.heading}",
                "fontWeight": "{font_weight.regular}",
                "lineHeight": "{line_height.sm}",
                "fontSize": "{font_size.sm}"
            },
            "type": "typography"
        }
    },
    "elevation": {
        "low": {
            "value": {
                "x": "0",
                "y": "0",
                "blur": "4px",
                "spread": "1px",
                "color": "rgba(35,57,92,0.1)",
                "type": "dropShadow"
            },
            "type": "boxShadow"
        },
        "medium": {
            "value": {
                "x": "0",
                "y": "0",
                "blur": "6px",
                "spread": "1px",
                "color": "rgba(35,57,92,0.12)",
                "type": "dropShadow"
            },
            "type": "boxShadow"
        },
        "high": {
            "value": {
                "x": "0",
                "y": "0",
                "blur": "8px",
                "spread": "1px",
                "color": "rgba(35,57,92,0.15)",
                "type": "dropShadow"
            },
            "type": "boxShadow"
        }
    },
    "size": {
        "xs": {
            "value": "16px",
            "type": "sizing"
        },
        "sm": {
            "value": "20px",
            "type": "sizing"
        },
        "md": {
            "value": "24px",
            "type": "sizing"
        },
        "lg": {
            "value": "32px",
            "type": "sizing"
        },
        "xl": {
            "value": "40px",
            "type": "sizing"
        },
        "2xl": {
            "value": "48px",
            "type": "sizing"
        },
        "3xl": {
            "value": "98px",
            "type": "sizing"
        }
    },
    "text_decoration": {
        "underline": {
            "value": "underline",
            "type": "textDecoration"
        }
    }
}