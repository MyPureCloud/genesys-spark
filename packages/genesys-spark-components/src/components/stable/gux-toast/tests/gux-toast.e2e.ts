import { E2EPage } from '@stencil/core/testing';
import { newSparkE2EPage, a11yCheck } from '../../../../test/e2eTestUtils';

async function clickDismissButton(page: E2EPage) {
  await page.evaluate(async () => {
    const element = document.querySelector('gux-toast');
    const dismissButtonElement =
      element.shadowRoot.querySelector('gux-dismiss-button');
    const button = dismissButtonElement.shadowRoot.querySelector('button');

    button.click();
  });
}

describe('gux-toast', () => {
  describe('#render', () => {
    [
      {
        description: 'should render success toast',
        html: `
          <div aria-live="polite">
            <gux-toast toast-type="success">
              <div slot="title">Success Example with Link</div>
              <div slot="message">This is an example message</div>
              <a slot="link" href="#">Link</a>
            </gux-toast>
          </div>
        `
      },
      {
        description: 'should render warning toast',
        html: `
          <div aria-live="polite">
            <gux-toast toast-type="warning">
              <div slot="title">Warning Example with Link</div>
              <div slot="message">This is an example message</div>
              <a slot="link" href="#">Link</a>
            </gux-toast>
          </div>
        `
      },
      {
        description: 'should render error toast',
        html: `
          <div aria-live="polite">
            <gux-toast toast-type="error">
              <div slot="title">Error Example with Link</div>
              <div slot="message">This is an example message</div>
              <a slot="link" href="#">Link</a>
            </gux-toast>
          </div>
        `
      },
      {
        description: 'should render info toast',
        html: `
          <div aria-live="polite">
            <gux-toast toast-type="info">
              <div slot="title">Info Example with Link</div>
              <div slot="message">This is an example message</div>
              <a slot="link" href="#">Link</a>
            </gux-toast>
          </div>
        `
      },
      {
        description: 'should render action toast',
        html: `
          <div aria-live="polite">
            <gux-toast toast-type="action">
              <gux-icon slot="icon" icon-name="fa/diamond-regular" decorative></gux-icon>
              <div slot="title">2 Actions</div>
              <div slot="message">This is an example message</div>
              <button slot="primary-button" type="button" onclick="notify(event)">
                Action 1
              </button>
              <button slot="secondary-button" type="button" onclick="notify(event)">
                Action 2
              </button>
            </gux-toast>
          </div>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-toast');
        await a11yCheck(page);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });

  describe('dismiss', () => {
    it('click dismiss button', async () => {
      const html = `
        <div aria-live="polite">
          <gux-toast lang="en" toastType="info">
            <div slot="message">This is the message</div>
          </gux-toast>
        </div>
      `;
      const page = await newSparkE2EPage({ html });
      const guxdismissSpy = await page.spyOnEvent('guxdismiss');
      const clickSpy = await page.spyOnEvent('click');

      expect(guxdismissSpy).not.toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-toast')).not.toBeNull();

      await clickDismissButton(page);
      await page.waitForChanges();

      expect(guxdismissSpy).toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-toast')).toBeNull();
    });

    it('click dismiss button and prevent default', async () => {
      const html = `
        <div aria-live="polite">
          <gux-toast lang="en" accent="info">
            <div slot="message">This is the message</div>
          </gux-toast>
        </div>
      `;
      const page = await newSparkE2EPage({ html });
      const guxdismissSpy = await page.spyOnEvent('guxdismiss');
      const clickSpy = await page.spyOnEvent('click');

      await page.evaluate(() => {
        document.addEventListener('guxdismiss', event => {
          event.preventDefault();
        });
      });

      expect(guxdismissSpy).not.toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-toast')).not.toBeNull();

      await clickDismissButton(page);
      await page.waitForChanges();

      expect(guxdismissSpy).toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-toast')).not.toBeNull();
    });
  });
});
