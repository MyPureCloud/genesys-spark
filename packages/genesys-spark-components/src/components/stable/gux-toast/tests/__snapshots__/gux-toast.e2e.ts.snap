// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-toast #render should render action toast 1`] = `"<gux-toast toast-type=\\"action\\" hydrated=\\"\\"> <gux-icon slot=\\"icon\\" icon-name=\\"fa/diamond-regular\\" decorative=\\"\\" size=\\"inherit\\" hydrated=\\"\\"></gux-icon> <div slot=\\"title\\">2 Actions</div> <div slot=\\"message\\">This is an example message</div> <button slot=\\"primary-button\\" type=\\"button\\" onclick=\\"notify(event)\\"> Action 1 </button> <button slot=\\"secondary-button\\" type=\\"button\\" onclick=\\"notify(event)\\"> Action 2 </button> </gux-toast>"`;

exports[`gux-toast #render should render error toast 1`] = `"<gux-toast toast-type=\\"error\\" hydrated=\\"\\"> <div slot=\\"title\\">Error Example with Link</div> <div slot=\\"message\\">This is an example message</div> <a slot=\\"link\\" href=\\"#\\">Link</a> </gux-toast>"`;

exports[`gux-toast #render should render info toast 1`] = `"<gux-toast toast-type=\\"info\\" hydrated=\\"\\"> <div slot=\\"title\\">Info Example with Link</div> <div slot=\\"message\\">This is an example message</div> <a slot=\\"link\\" href=\\"#\\">Link</a> </gux-toast>"`;

exports[`gux-toast #render should render success toast 1`] = `"<gux-toast toast-type=\\"success\\" hydrated=\\"\\"> <div slot=\\"title\\">Success Example with Link</div> <div slot=\\"message\\">This is an example message</div> <a slot=\\"link\\" href=\\"#\\">Link</a> </gux-toast>"`;

exports[`gux-toast #render should render warning toast 1`] = `"<gux-toast toast-type=\\"warning\\" hydrated=\\"\\"> <div slot=\\"title\\">Warning Example with Link</div> <div slot=\\"message\\">This is an example message</div> <a slot=\\"link\\" href=\\"#\\">Link</a> </gux-toast>"`;
